it/src/main/java/com/google/cloud/teleport/it/spanner/SpannerResourceManager.java

```diff
@@ -0,0 +1,79 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.CreateDatabaseException;
+import com.google.cloud.teleport.it.spanner.SpannerResourceManagerException.UpdateDatabaseException;
+
+/** Interface for managing spanner resources in integration tests. */
+public interface SpannerResourceManager {
+
+  /**
+   * Creates a table given a CREATE TABLE DDL statement.
+   *
+   * @param statement The CREATE TABLE DDL statement.
+   * @throws CreateDatabaseException if there is an issue creating the database with the table.
+   * @throws IllegalStateException if method is called after resources have been cleaned up.
+   * @throws UpdateDatabaseException if there is an issue creating the table.
+   */
+  void createTable(String statement)
+      throws CreateDatabaseException, IllegalStateException, UpdateDatabaseException;
+
+  /**
+   * Inserts a given record into a table.
+   *
+   * @param tableRecord A mutation object representing the table record.
+   * @throws IllegalStateException if method is called after resources have been cleaned up.
+   */
+  void insertTableRecord(Mutation tableRecord) throws IllegalStateException;
+
+  /**
+   * Inserts a collection of table records into a table.
+   *
+   * @param tableRecords A collection of mutation objects representing table records.
+   * @throws IllegalStateException if method is called after resources have been cleaned up.
+   */
+  void insertTableRecords(Iterable<Mutation> tableRecords) throws IllegalStateException;
+
+  /**
+   * Reads all the rows in a table.
+   *
+   * @param tableId The id of the table to read rows from.
+   * @param columnNames The table's column names.
+   * @return A ResultSet object containing all the rows in the table.
+   * @throws IllegalStateException if method is called after resources have been cleaned up.
+   */
+  ResultSet readTableRecords(String tableId, String... columnNames) throws IllegalStateException;
+
+  /**
+   * Reads all the rows in a table.
+   *
+   * @param tableId The id of table to read rows from.
+   * @param columnNames A collection of the table's column names.
+   * @return A ResultSet object containing all the rows in the table.
+   * @throws IllegalStateException if method is called after resources have been cleaned up.
+   */
+  ResultSet readTableRecords(String tableId, Iterable<String> columnNames)
+      throws IllegalStateException;
+
+  /**
+   * Deletes all created resources (instance, database, and tables) and cleans up all Spanner
+   * sessions, making the Spanner and manger objects unusable.
```

Comment:
Don't mention the  Spanner object. That's an implementation detail. Users only care if the resource manager is unusable.